### PR TITLE
android build support on macOS

### PR DESCRIPTION
**Pull request by nnkalita**
added android build support on macOS

### CODE REVIEW

File: android/build-kaldi.sh [hunk 1]
```diff
@@ -30,13 +30,14 @@ if [ ! -d "$ANDROID_SDK_HOME/ndk-bundle" ]; then
 fi
 
 set -x
-
+unameOS1="$(uname -s)"
+unameOS=`echo $unameOS1| tr '[:upper:]' '[:lower:]'`
```
Comment (lpierron) [inline android/build-kaldi.sh:34]: Why not:
```bash
unameOS="$(uname -s | tr '[:upper:]' '[:lower:]')"
```
or replacing `$unameOS` by `$OSTYPE` predefined in `bash`.

**Review by lpierron — CHANGES_REQUESTED** [review]

File: android/build-kaldi.sh [hunk 2]
```diff
@@ -45,13 +46,19 @@ mkdir -p $WORKDIR_ARM64/local/lib $WORKDIR_ARM32/local/lib $WORKDIR_X86_64/local
...
+    darwin*)    sedCmd=gsed;;   
+    *)          sedCmd=sed
+esac
+$sedCmd -i 's/APP_STL := gnustl_static/APP_STL := c++_static/g' jni/Application.mk && \
+$sedCmd -i 's/android-10/android-21/g' project.properties && \
+$sedCmd -i 's/APP_ABI := armeabi armeabi-v7a/APP_ABI := armeabi-v7a arm64-v8a x86_64/g' jni/Application.mk && \
+$sedCmd -i 's/LOCAL_MODULE:= testlapack/#LOCAL_MODULE:= testlapack/g' jni/Android.mk && \
+$sedCmd -i 's/LOCAL_SRC_FILES:= testclapack.cpp/#LOCAL_SRC_FILES:= testclapack.cpp/g' jni/Android.mk && \
+$sedCmd -i 's/LOCAL_STATIC_LIBRARIES := lapack/#LOCAL_STATIC_LIBRARIES := lapack/g' jni/Android.mk && \
+$sedCmd -i 's/include $(BUILD_SHARED_LIBRARY)/#include $(BUILD_SHARED_LIBRARY)/g' jni/Android.mk && \
```
Comment (lpierron) [inline android/build-kaldi.sh:61]: I think, it will be better to replace `sed -i ` by `sed -i.bak -e `.

**Review by lpierron — CHANGES_REQUESTED** [review]

File: android/build-kaldi.sh (hunk 1 above)
Comment (lpierron) [reply android/build-kaldi.sh:34]: `$OSTYPE` gives the release number, so maybe it's too specific.

File: android/build-kaldi.sh (hunk 2 above)
Comment (nnkalita) [reply android/build-kaldi.sh:61]: Thanks for your feedback, I have updated the script.

**Pull request closed without merging**